### DISCUSSION TITLE
Why does `flutter run` want to put a binary in a root directory?

### DISCUSSION THREAD

**Discussion by spikespaz** [fzyzcjy/flutter_rust_bridge#371] [Q&A]
I am getting the following output from `flutter run`.

```
   Compiling native v0.1.0 (/home/jacob/Documents/github.com/spikespaz/umapacker/flutter_rust_bridge_template/native)
    Finished dev [unoptimized + debuginfo] target(s) in 3.74s
CMake Error at cmake_install.cmake:66 (file):
  file INSTALL cannot copy file
  "/home/jacob/Documents/github.com/spikespaz/umapacker/flutter_rust_bridge_template/build/linux/x64/debug/intermediates_do_not_run/flutter_rust_bridge_template"
  to "/usr/local/flutter_rust_bridge_template": Permission denied.


Building Linux application...                                           
Exception: Build process failed
```

I don't want this to do *anything* with system directories. No no.

**Comment by fzyzcjy** [comment]
Sounds like it is a problem with Flutter instead of the bridge. Maybe ask on StackOverflow or flutter/flutter github repo?

/cc @Desdaemon since this template is created by him

**Reply by spikespaz** [reply]
I'd be beaten on StackOverflow because there's nothing minimal about this example.

**Reply by fzyzcjy** [reply]
What about firstly create a brand new flutter project and play with it and see whether it errors like this

**Comment by Desdaemon** [comment]
This seems to be a [weird bug](https://github.com/flutter/flutter/issues/59890) with Flutter Linux. Can you try running `flutter clean` and see if it works?

**Reply by spikespaz** [reply]
I will revisit this when I try to integrate flutter with a project again. for now I have abandoned the GUI and am working on the core/CLI of my program. This issue should stay open, and when I do run into the issue again (or if I don't) I will leave a comment here.

@fzyzcjy I have tried with a new Flutter project and the issue does not occur. Nor does it happen when I try (just now) to run using the example from quickstart. I had this issue when I was trying to build the project up myself for educational purposes.

**Reply by fzyzcjy** [reply]
Sure, take your time!

**Reply by spikespaz** [reply]
`flutter clean` did the trick. Strange, because it should have already been a clean environment. Also, any hope for simplifying the template or creating another one that is proper MCVE?

**Comment by spikespaz** [comment]
Why is `flutter_rust_bridge_codegen` failing to guess my crate?

**Reply by fzyzcjy** [reply]
Please raise a separate issue with enough logs

**Comment by yooniss** [comment]
flutter clean should do
